lib/cher/error.go

```diff
@@ -153,13 +154,34 @@ func (e E) Value() (driver.Value, error) {
 	return json.Marshal(e)
 }
 
+// WrapIfNotCher will not wrap the error if it is any cher except unknown.
+//
+// Unless you don't know the codes you could get back, you should use WrapIfNotCherCodes.
 func WrapIfNotCher(err error, msg string) error {
 	if err == nil {
 		return nil
 	}
 
 	var cErr E
 	if errors.As(err, &cErr) {
+		if cErr.Code == Unknown {
+			return errors.Wrap(err, msg)
```

Comment:
Are there any cases where it could be `unknown`, but have reasons or metadata that we'd want to keep? I can't think of any but I'm not 100% sure.

Reply:
I wouldn't have thought so, and if they do they deserve to go. Why don't reasons or metadata get serialised into error strings?